pkg/handler/handler.go

```diff
@@ -181,6 +204,8 @@ func (o *OpenAPIService) RegisterOpenAPIVersionedService(servePath string, handl
 					contentType := accepts.Type + "/" + accepts.SubType
 					w.Header().Set("Content-Type", contentType)
 
+					lastModified := etagCache.LastModified(etag)
```

Comment:
I was expecting both content types to have distinct etags, but this code path (which can serve either one) clobbers the time and existing etag either way.

But I think I'm wrong and the etag is shared by both formats. Is that intentional?

Reply:
The tag is shared by both formats, I didnt write the original code, but I think it is intentional. The e-tag does not necessarily depend upon the byte value of the response.

[From MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag#directives):
>  The method by which ETag values are generated is not specified. Typically, the ETag value is a hash of the content, a hash of the last modification timestamp, or just a revision number. 

I do think it is confusing that the per-content-type `GetDataAndEtag()` returns an etag, but the etag doesn't vary according to content type.

 Given this is now an invariant now being required by the etagcache the new code might be more error prone: alternating requests of each content type would keep bumping `lastModified` if the invariant was violated.